File: backoffice/src/product/domain/valueObjects/ProductId.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain                      import StringValueObject
from ..exceptions.InvalidProductIdException import InvalidProductIdException

"""
 *
 * Class 
 *
"""

class ProductId( StringValueObject ):

    """
     *
     * Methods 
     *
    """
    
    def __init__( self, value : str ) -> None:
        super().__init__( value )
        if self.isEmpty():
            raise InvalidProductIdException( id )

File: preparation/src/product/domain/valueObjects/ProductId.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.product.domain {

    public class ProductId : UuidValueObject {

        /* 
         * 
         * Methods
         *
        */

        public ProductId( string value ) : base( value ) {
        }

    }

}